contracts/types/PoolId.sol

```diff
@@ -7,7 +7,9 @@ type PoolId is bytes32;
 
 /// @notice Library for computing the ID of a pool
 library PoolIdLibrary {
-    function toId(PoolKey memory poolKey) internal pure returns (PoolId) {
-        return PoolId.wrap(keccak256(abi.encode(poolKey)));
+    function toId(PoolKey memory poolKey) internal pure returns (PoolId poolId) {
+        assembly("memory-safe") {
+            poolId := keccak256(poolKey, mul(32, 5))
```

Comment:
maybe a code comment above the assembly block could be nice to save a maintainer/reader a minute
```
// PoolId.wrap(keccak256(abi.encode(poolKey)));
```

Reply:
This and next comment addressed in 0f6c44850ae5eb68078cb5308940b71d6ff11b03!